### DISCUSSION TITLE
atomWithStorage glitches

### DISCUSSION THREAD

**Discussion by larini** [pmndrs/jotai#2675] [Q&A]
I defined an atom using windows.localStorage, and it works very well:

	import { atom } from "jotai";

	const useSidebarAtom = atom(
		window.localStorage.getItem("sidebarCollapsed") === "true",
		(get, set) => {
			const value = !get(useSidebarAtom);
			set(useSidebarAtom, value);
			window.localStorage.setItem("sidebarCollapsed", value);
	  });

	export { useSidebarAtom };

but when trying to use a atomWithStorage instead, the UI seams to update twice. 

	import { atomWithStorage } from "jotai/utils"
	const useSidebarAtom = atomWithStorage("sidebarCollapsed", true);
	export { useSidebarAtom };

Let's suppose that the local storage value for this variable is "false".
When using atomWithStorage, looks like the sidebar is collapsed and then not.

**Comment by rothsandro** [comment]
`atomWithStorage` initializes with the default value. You can change this via `getOnInit` option: https://jotai.org/docs/utilities/storage#parameters

**Reply by larini** [reply]
> `atomWithStorage` initializes with the default value. You can change this via `getOnInit` option: https://jotai.org/docs/utilities/storage#parameters

Unfortunately I got the same behavior using `getOnInit` option.

    import { atomWithStorage } from "jotai/utils"
    const useSidebarAtom = atomWithStorage("sidebarCollapsed", { getOnInit: true });
    export { useSidebarAtom };

**Reply by rothsandro** [reply]
Can you provide a minimal reproduction?
